File: bench.py
import random, math
def bench_monte_carlo_py(n_sim_points):
    n_in = 0
    for i in range(0,n_sim_points):
        x = random.uniform(-0.5, 0.5)
        y = random.uniform(-0.5, 0.5)
        if math.sqrt(x**2 + y**2) <= 0.5:
            n_in +=1
    
    return 4*n_in/n_sim_points

if __name__ == "__main__":
    print(f"pi is approx {bench_monte_carlo_py(1_000_000)}")

File: bench.cpp
#include <iostream>
#include <random>

double bench_monte_carlo_pi_c(uint64_t n_sim_points) {
  uint64_t n_in = 0;
  auto gen = std::default_random_engine();
  auto distribution = std::uniform_real_distribution<double>(-0.5, 0.5);
  for (size_t i = 0; i < n_sim_points; i++) {
    double x = distribution(gen);
    double y = distribution(gen);
    if (sqrt(x * x + y * y) <= 0.5) {
      n_in++;
    }
  }
  return 4 * double(n_in) / n_sim_points;
}

int main() { std::cout << bench_monte_carlo_pi_c(1000000) << std::endl; }